core/src/main/scala/org/apache/spark/deploy/master/FileSystemPersistenceEngine.scala

```diff
@@ -58,7 +58,7 @@ private[spark] class FileSystemPersistenceEngine(
     if (!created) { throw new IllegalStateException("Could not create file: " + file) }
     val serializer = serialization.findSerializerFor(value)
     val serialized = serializer.toBinary(value)
-    val out = new FileOutputStream(file)
+    val out = new BufferedOutputStream(new FileOutputStream(file))
```

Comment:
I agree. This doesn't help but doesn't really hurt either since we're only allocating 8K. Not a huge deal but I might leave this one out.